pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -1,339 +1,6 @@
-| [1, 3]        | list      | [1, 3]              | GPUs 1 and 3                    |
-+---------------+-----------+---------------------+---------------------------------+
-| "0"           | str       | [0]                 | GPU 0                           |
-+---------------+-----------+---------------------+---------------------------------+
-| "3"           | str       | [3]                 | GPU 3                           |
-+---------------+-----------+---------------------+---------------------------------+
-| "1, 3"        | str       | [1, 3]              | GPUs 1 and 3                    |
-+---------------+-----------+---------------------+---------------------------------+
-| "-1"          | str       | [0, 1, 2, ...]      | all available GPUs              |
-+---------------+-----------+---------------------+---------------------------------+
-
-
-Multi-node
-----------
-
-Multi-node training is easily done by specifying these flags.
-
-.. code-block:: python
-
-    # train on 12*8 GPUs
-    trainer = Trainer(gpus=8, num_nodes=12, distributed_backend='ddp')
-
-
-You must configure your job submission script correctly for the trainer to work.
- Here is an example script for the above trainer configuration.
-
-.. code-block:: bash
-
-    #!/bin/bash -l
-
-    # SLURM SUBMIT SCRIPT
-    #SBATCH --nodes=12
-    #SBATCH --gres=gpu:8
-    #SBATCH --ntasks-per-node=8
-    #SBATCH --mem=0
-    #SBATCH --time=0-02:00:00
-
-    # activate conda env
-    conda activate my_env
-
-    # -------------------------
-    # OPTIONAL
-    # -------------------------
-    # debugging flags (optional)
-    # export NCCL_DEBUG=INFO
-    # export PYTHONFAULTHANDLER=1
-
-    # PyTorch comes with prebuilt NCCL support... but if you have issues with it
-    # you might need to load the latest version from your  modules
-    # module load NCCL/2.4.7-1-cuda.10.0
-
-    # on your cluster you might need these:
-    # set the network interface
-    # export NCCL_SOCKET_IFNAME=^docker0,lo
-    # -------------------------
-
-    # random port between 12k and 20k
-    export MASTER_PORT=$((12000 + RANDOM % 20000))
-
-    # run script from above
-    python my_main_file.py
-
-.. note:: When running in DDP mode, any errors in your code will show up as an NCCL issue.
- Set the `NCCL_DEBUG=INFO` flag to see the ACTUAL error.
-
-Normally now you would need to add a distributed sampler to your dataset, however
-Lightning automates this for you. But if you still need to set a sampler Lightning will
-not interfere nor automate it.
-
-Here's an example of how to add your own sampler (again no need with Lightning).
-
-.. code-block:: python
-
-    # ie: this:
-    dataset = myDataset()
-    dataloader = Dataloader(dataset)
-
-    # becomes:
-    dataset = myDataset()
-    dist_sampler = torch.utils.data.distributed.DistributedSampler(dataset)
-    dataloader = Dataloader(dataset, sampler=dist_sampler)
-
-
-Auto-slurm-job-submission
```

Comment:
what about this? @awaelchli

Reply:
I moved it and renamed the title to "Building SLURM scripts" because auto slurm job submission was already described in the other document and this title seemed to fit better.

Reply:
I found it later too :]